ApollosProject/apollos-apps
# feat: added more info to content single and analytics for buttons

This PR adds potential parent node info to content single analytics and also tracks button presses. I tried to follow similar patterns already in the code base. 

~~@vinnyjth or @redreceipt I could use a hand with testing these out in Amplitude, and I'm going to mark this as a draft PR in case we want to adjust how I have gone about things with this one.~~

## Screenshot
Proof that the events are coming through to Amplitude
<img width="821" alt="Screen Shot 2021-12-02 at 10 55 39 AM" src="https://user-images.githubusercontent.com/72768221/144468540-d0ae9930-1e2e-420a-a050-4055a36e7a69.png">

## Testing
1. Open Flipper
1. Open a content item with a button in your simulator
1. Click the button in the simulator
1. Check your network requests in Flipper (search for trackEvent)
![Screen Shot 2021-12-02 at 11 05 42 AM](https://user-images.githubusercontent.com/72768221/144469036-5cac859c-4c61-48a2-ad34-28692d246184.png)
1. Select the most recent request in the filtered list
1. You should see a response like this if the event was tracked
![Screen Shot 2021-12-02 at 11 05 29 AM](https://user-images.githubusercontent.com/72768221/144469006-989a6376-82b6-4628-b609-d3eed9e3b550.png)
